agent/grpc/public/services/peerstream/replication.go

```diff
@@ -240,12 +241,48 @@ func (s *Server) handleUpdateService(
 	sn structs.ServiceName,
 	pbNodes *pbservice.IndexedCheckServiceNodes,
 ) error {
+	_, peering, err := s.GetStore().PeeringRead(nil, state.Query{Value: peerName})
+	if err != nil {
+		return fmt.Errorf("failed to read peering: %w", err)
+	}
+	// an invariant here should be that a peering exists for PeeringRead
+	if peering == nil {
+		return fmt.Errorf("did not find peering %q", peerName)
+	}
+
+	logger := s.Logger.With("peerName", peerName, "peerID", peering.ID)
+
 	// Capture instances in the state store for reconciliation later.
 	_, storedInstances, err := s.GetStore().CheckServiceNodes(nil, sn.Name, &sn.EnterpriseMeta, peerName)
 	if err != nil {
 		return fmt.Errorf("failed to read imported services: %w", err)
 	}
 
+	// account for imported services
+	if len(storedInstances) > 0 && len(pbNodes.GetNodes()) < 1 {
+		// decrement the service count as the service was removed/ is being removed
+		logger.Trace("decrementing imported services count", "serviceName", sn.String())
+
+		id := peering.GetID()
+		st, err := s.Tracker.MutableStreamStatus(id)
+		if err != nil {
+			logger.Trace("failed to find stream status, abandoning decrement for imported services count", "serviceName", sn.String())
```

Comment:
I don't think that logging an error and moving forward is appropriate here

Take a look at what leads to creating/deleting entries in the stream tracker

Reply:
that's fair; erroring out now